src/App.tsx

```diff
@@ -15,48 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+type Reset = (
+  handleSorted: (sorted: SortType) => void,
+  handleReversed: (isRevers: boolean) => void,
+) => void;
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      default: return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsRevesed] = useState(false);
+  const visibleGoods = getReorderedGoods(
+    goodsFromServer, { sortType, isReversed },
+  );
+
+  const handleSort = (typeSort: SortType) => setSortType(typeSort);
+  const handleReverse = (isRevers: boolean) => setIsRevesed(isRevers);
+  const reset: Reset = (handleSorted, handleReversed) => {
+    handleSorted(SortType.NONE);
+    handleReversed(false);
+  };
```

Comment:
too complicated, just this way 🙂 
```suggestion
  const handleSort = (type: SortType) => {
    setSortType(type);
  };
  
  const handleReverse = () => {
    setIsRevesed(!isReverse);
  };
  
  const handleReset = () => {
    setSortType(SortType.NONE);
    setIsRevesed(false);
  };
```

Reply:
Done, but for the reset function I spent a half hour,